app/components/viral/modal_component.html.erb

```diff
@@ -0,0 +1,99 @@
+<div data-controller="viral--modal-component">
+  <!-- Modal toggle -->
+  <button
+    data-viral--modal-component-target="button"
+    data-action="click->viral--modal-component#open"
+    class="
+      block
+      text-white
+      bg-blue-700
+      hidden
+      hover:bg-blue-800
+      focus:ring-4
+      focus:outline-none
+      focus:ring-blue-300
+      font-medium
+      rounded-lg
+      text-sm
+      px-5
+      py-2.5
+      text-center
+      dark:bg-blue-600
+      dark:hover:bg-blue-700
+      dark:focus:ring-blue-800
```

Comment:
**nitpick (blocking):** For now just used the default `btn btn-default` classes if you want to make the default button or update all the `*-blue* to be *-primary* so it grabs the primary colour from the configuration file.